pkg/store/bucket.go

```diff
@@ -742,15 +738,35 @@ func blockSeries(
 			})
 			s.refs = append(s.refs, meta.Ref)
 		}
-		if len(s.chks) > 0 {
-			if err := chunksLimiter.Reserve(uint64(len(s.chks))); err != nil {
-				return nil, nil, errors.Wrap(err, "exceeded chunks limit")
+		if find || len(s.chks) > 0 {
+			// Reserve chunksLimiter if we save chunks.
+			if len(s.chks) > 0 {
```

Comment:
I think moving this outside the enclosing `if` could make it more readable.

Reply:
Thanks. This makes sense!